2017/proposals/ranasinghe-kanchana.md

```diff
@@ -0,0 +1,104 @@
+The work of this project will stem from existing development done categorical color support in Matplotlib issues [6889](https://github.com/matplotlib/matplotlib/pull/6889) and [7383](https://github.com/matplotlib/matplotlib/issues/7383). The starting point for the project will be extending the imshow function (matplotlib.axes._axes.imshow) to support categorical data. This mainly focuses on providing it unit knowledge with regards to non-numerical data. The approach followed here will be the initial basis for this extended support for categorical data. It will be used as a foundation to build the heat-maps API, and normalization and color-map APIs. 
+Currently, the process of using matplotlib.pyplot for heat-maps and other scalarMappables includes the creation of a color-map followed by normalization of all available data to fit this color-map. Color-maps plotting functionality is threefold: sequential, divergent and qualitative. Considering the intrinsic discreteness of categorical data, the extension of the color-maps API will focus on its qualitative plotting ability. However the fundamental behavior of the color-maps API will require changes for handling categorical data as it is currently dependent on the presence of numerical scalar values in the data to be plotted. 
+With regards to the normalization function, since this extension will require working with categorical data, the concept of normalization of data seems to be rendered useless. However, the possibility of integrating this into the APIs to be built will be evaluated. Also during the course of the project focus will be given to the currently available discrete normalization methods (matplotlib.colors.BoundaryNorm) which handle data in a different sense. 
+Given this functionality, the final task is entwining all this functionality to allow the plotting of heat-maps and other scalarMappables. This will focus on providing units support for the existing functions with regards to categorical data, followed by integration of the existing frameworks (including those built previously in this project) to better support categorical data visualization. 
+The completion of this project will require the creation of certain new classes for categorical data handling as well as extending of currently available classes and APIs.  
+
+## Schedule of Deliverables
+
+### May 1th - May 28th, **Community Bonding Period**
+* Working on categorical data support for imshow function 
+* Self-implementation of previous work done to support categorical color for better understanding
+* Setting up of project blog covering all major technical details
+* Working on MEPs in Matplotlib Developers’ Guide, and Matplotlib issues related to normalization, color-map, and color-bar code. 
+
+### May 29th - June 3rd
+* imshow functionality
```

Comment:
I strongly suggest allocating way way more time for this task, like a week on each bullet point at least.